### PR TITLE
Fix/checkbox and datepicker

### PR DESCRIPTION
**Pull request by jsebrech**
## PR Checklist

This PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] A changelog entry has been added to CHANGELOG.md if necessary

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] react application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #246 #247 #248 #249 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

Dependencies were cleaned up and some were moved to peer dependencies. Should not be a breaking change, because no new dependencies were added.

## Resolved issues

Fixes: #246
Fixes: #247
Fixes: #248
Fixes: #249

### CODE REVIEW

File: packages/form/src/Checkbox/Checkbox.js [hunk 1]
```diff
@@ -22,6 +22,8 @@ class Checkbox extends Component<Props> {
     }
   }
 
+  defaultId = 'aui-cb-' + Math.random().toString(36);
```
Comment (TriangleJuice) [inline packages/form/src/Checkbox/Checkbox.js:25]: Misschien beter met [defaultProps](https://reactjs.org/docs/typechecking-with-proptypes.html#default-prop-values) werken hier?

File: packages/form/src/Datepicker/Datepicker.js [hunk 2]
```diff
@@ -84,25 +84,27 @@ class Datepicker extends Component<Props> {
     disabled: false,
   };
 
+  defaultId = 'aui-date-' + Math.random().toString(36);
```
Comment (TriangleJuice) [inline packages/form/src/Datepicker/Datepicker.js:87]: Zelfde comment als hierboven.

File: packages/form/src/Datepicker/Datepicker.spec.js [hunk 3]
```diff
@@ -30,7 +30,7 @@ describe('Datepicker', () => {
 	});
 
 	it('should set the data-qa attribute', () => {
-		const datepicker = shallow(<Datepicker qa="id-1234" />);
+		const datepicker = shallow(<Datepicker qa="id-1234" id="id-4321" />);
```
Comment (TriangleJuice) [inline packages/form/src/Datepicker/Datepicker.spec.js:33]: Zelfde vraag als hierboven.

File: packages/hero/src/Hero.spec.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import Hero from './Hero';
+
+describe('Hero', () => {
```
Comment (TriangleJuice) [inline packages/hero/src/Hero.spec.js:6]: Thanks for the first test 👍

**Review by TriangleJuice — CHANGES_REQUESTED** [review]
Merci, Joeri! Ziet er al goed uit. Nog enkele kleine dingetjes voor na je vakantie.

**Review by vademo — APPROVED** [review]
Paar goede opmerkingen van @TriangleJuice maar ziet er voor de rest goed uit. 
excuses voor de late review

File: packages/form/src/Checkbox/Checkbox.spec.js [hunk 5]
```diff
@@ -9,7 +9,7 @@ describe('Checkbox', () => {
     });
 
     test('Checkbox should contain a data-qa attribute', () => {
-      const component = shallow(<Checkbox qa="id-1234" />);
+      const component = shallow(<Checkbox qa="id-1234" id="id-4321" />);
```
[the remark this answers is not in the document]
Comment (jsebrech) [reply packages/form/src/Checkbox/Checkbox.spec.js:12]: dit gaf een code coverage verbetering

File: packages/form/src/Checkbox/Checkbox.js (hunk 1 above)
Comment (jsebrech) [reply packages/form/src/Checkbox/Checkbox.js:25]: Defaultprops zijn per-component ipv per-instance. Een id moet uniek zijn per instance.

File: packages/form/src/Datepicker/Datepicker.js (hunk 2 above)
Comment (jsebrech) [reply packages/form/src/Datepicker/Datepicker.js:87]: Zelfde antwoord als hierboven :)

File: packages/form/src/Datepicker/Datepicker.spec.js (hunk 3 above)
Comment (jsebrech) [reply packages/form/src/Datepicker/Datepicker.spec.js:33]: Zelfde antwoord als hierboven :)

**Review by TriangleJuice — APPROVED** [review]

**Pull request merged**